File: python/java_to_python/07-pgmThread/src/co/edu/campusucc/poo/ThreadRunnable.py
import threading
import time

class ThreadRunnable(threading.Thread):
    def __init__(self):
        threading.Thread.__init__(self)
        self.times = 0

    def run(self):
        print("Runnable Thread Started Running...on UCC University")
        print("Random Times: " + str(self.times))
        try:
            self.printTimes()
        except Exception as e:
            print(e)

    # Defined Behavior
    def printTimes(self):
        timesCount = 0
        while timesCount <= self.times:
            print("-", end="")
            timesCount += 1
            time.sleep(0.05)
        print("> OK")

    # Getters and Setters
    def getTimes(self):
        return self.times

    def setTimes(self, times):
        self.times = times

File: ruby/pgms_ruby/07-pgmThread/ThreadRunnable.rb
require 'thread'

class ThreadRunnable < Thread
  def initialize
    super
    @times = 0
  end

  def run
    puts "Runnable Thread Started Running...on UCC University"
    puts "Random Times: #{@times}"
    begin
      printTimes
    rescue Exception => e
      puts e
    end
  end

  # Defined Behavior
  def printTimes
    timesCount = 0
    while timesCount <= @times
      print "-"
      timesCount += 1
      sleep(0.05)
    end
    puts "> OK"
  end

  # Getters and Setters
  def getTimes
    @times
  end

  def setTimes(times)
    @times = times
  end
end